pkg/apis/extensions/v1beta1/types.go

```diff
@@ -450,6 +454,8 @@ type JobConditionType string
 const (
 	// JobComplete means the job has completed its execution.
 	JobComplete JobConditionType = "Complete"
```

Comment:
Complete is an adjective, and so are other ConditionTypes, such as Healthy and Ready.  Fail is a verb (or sometimes a noun).   Failed is the adjective form.  Suggest using Failed.  @bgrant0607 to confirm